src/Domain.jl

```diff
@@ -392,37 +392,46 @@ end
 """
     site_selection(domain::Domain, criteria::DataFrame, area_to_seed::Float64, ts::Int, n_reps::Int, alg_ind::Int)
 
+# Arguments
+- domain : ADRIA Domain type, indicating geographical domain to perform site selection over.
+- criteria : DataFrame of criteria weightings and thresholds (can be a scenario DataFrame).
+- area_to_seed : area of coral to be seeded at each time step in km^2
+- ts : time step at which seeding and/or shading is being undertaken.
+- n_reps : number of dhw/wave replicates to use in site selection.
+- scen_ind : scenario id for criteria (if using a scenario DataFrame, otherwise use scen_id = 1).
+- alg_ind : MCDA algorithm to use in site selection.
+
 # Returns
-Matrix : n_reps * sites * 3
-last dimension indicates: site_id, seeding rank, shading rank
+- Matrix : n_reps * sites * 3 (last dimension indicates: site_id, seeding rank, shading rank)
 """
-function site_selection(domain::Domain, criteria::DataFrame, area_to_seed::Float64, ts::Int, n_reps::Int, alg_ind::Int)
+function site_selection(domain::Domain, criteria::DataFrame, area_to_seed::Float64, ts::Int, n_reps::Int, scen_ind::Int, alg_ind::Int)
```

Comment:
I'm not clear on what `ts` and `scen_ind` are for? 

They don't seem to be used.

Reply:
Sorry, just saw `scen_ind`, but still can't see `ts`

Reply:
ts is used to index the dhw/wave time step, I added scen_ind in case we wanted to have a scanario dataframe, generated using the sampler, and then run site selectionon scenarios from the dataframe (indicated using the scen_ind index).

Reply:
Okay I see now - hard to manually look through because a straightforward `find` (ctrl+f) on `ts` would pick up too many false matches.

How is coral state at time $ts$ incorporated, or is that out of scope of concerns in this use case?

I would split the function up so you have one that handles scenario specific cases, and a more general one that accepts the scenario spec.

e.g., something like:

```julia
function site_selection(domain, criteria, ....)
    # some set up code
    # calls the scenario specific site selection function
    for scen_criteria in eachrow(criteria)
        some_result_variable = site_selection(scen_criteria, ......)
        
        # do something with results....
    end
end

function site_selection(criteria::DataFrameRow, ....)
     # do the thing ....
end
```

Every time you use an index you're introducing the risk that the wrong index gets used, and also increase maintenance burden cos every time something changes you have to go back and update all the places that index was used and make sure it was all updated correctly.

In this function alone you have to touch 15+ lines if something changes...

Reply:
Ok cool, I'll work on making these changes tomorrow. The current version of site_selection uses whatever is in domain.init_coral_cover for cover, but this should probably be changed to a separate input instead of using domain so it can be easily integrated with ADRIA's time dynamic model...

Reply:
I'm just working on implementing this and I'm having issues importing the DMCDA_vars variable type into Domain - would I use "import ADRIA:dMCDA" to import this ? It also makes me think though - should site_selection be in the dMCDA.jl file instead of Domain.jl?

Reply:
> I'm having issues importing the DMCDA_vars variable type into Domain

I don't quite follow, sorry. Could you show me what you're trying to do? `dMCDA_vars` is already used inside Domain so I'm not sure how you're trying to use it.

> should site_selection be in the dMCDA.jl file instead of Domain.jl?

Probably, particularly if the function is MCDA specific. I think the current structure needs a review - too many hangovers from MATLAB. As I said, I was trying to keep things matching as much as possible to avoid the perception that I made wild changes.

Reply:
Sorry, I forget ther's a few things called MCDA ... I mean the variable struct called MCDA_vars, I want to use it so there's less inputs in the function called in the scenario loop. But if you think it's fine to move the site selection functions to dMCDA.jl then I can do that and DMCDA_vars is already defined there so I don't have to import it.

Reply:
actually I just realised we were using it before and it wasn't complaining about it being undefined, maybe something else is going wrong... but anyway I can move it to dMCDA.jl